### DISCUSSION TITLE
Is Unified kernel image suppported for arm64?

### DISCUSSION THREAD

**Discussion by gp-sbg** [uapi-group/specifications#104] [Q&A] [answered]
I booted up an ubuntu arm machine in OCI and made unified kernel image using the following sbctl command,

`sudo ./sbctl bundle -e /usr/lib/systemd/boot/efi/linuxaa64.efi.stub -f /boot/initrd.img-6.5.0-1020-oracle -k /boot/vmlinuz-6.5.0-1020-oracle linux_arm.efi`

I copied the unified kernel image so created into /boot and in the grub, i did 

`chainloader (hd0,gpt1)/boot/linux_arm.efi 
boot`

But it fails in grub telling

**kern/verifiers.c:214: string: boot, type: 2
Execution of embedded linux image failed: Load Error
error: unknown error.**

I tried the exact same thing from an ubuntu x86 machine and the unified kernel image created in x86 is working correctly.

So, is unified kernel image not supported in arm64??

I also ensured that my kernel has the CONFIG_EFI_ZBOOT=y as mentioned in https://wiki.gentoo.org/wiki/Unified_kernel_image#Supported_architectures

**Comment by Foxboron** [comment, marked as answer]
Please don't use `sbctl` to make UKIs as it will do a worse job than using `ukify` or any of the proper initramfs generators.

**Comment by gp-sbg** [comment]
i am using systemd version 233, whereas systemd-unkify is only present in systemd version 255. I am also unable to download and build ukify manually, that's why i had gone down the sbctl route, since i can just download the sbctl repo and then run a make command, and the sbctl tool is ready to use.
Can u kindly tell me how to install ukify without upgrading systemd version?

**Reply by daandemeyer** [reply]
@gp-sbg You need a very recent systemd version and kernel to use UKIs on arm64. Please switch to a modern system with systemd 254 or newer. ukify will also be available then.

**Reply by poettering** [reply]
v233 is more than 7 years old. That's *pre-historic*
